plugins_src/import_export/wpc_svg_path.erl

```diff
@@ -41,34 +65,259 @@
+            [{key,transforms_in_layer_nms}]},
+       panel,
+       {hframe,[info_button()]},
+       {hframe,[{label,?__(3,
+           "(For best results when importing from Inkscape,\n"
+           "convert all objects to paths and save as a plain .svg)")}]}
+       ],
     wpa:dialog(Ask, ?__(1,".svg Path Import Options"),
         Dialog,
         fun(Res) -> {file,{import, svg, Res}} end);
-command({file,{import, svg, [Nsub]}}, St) ->
-    Props = [{extensions,[{".svg",?__(4,".svg File")}]}],
-    wpa:pref_set(wpc_svg_path, svg_bisections, Nsub),
-    wpa:import(Props, fun(F) -> make_svg(F, Nsub) end, St);
+command({file,{import, svg, List}}, St) when is_list(List) ->
+    Nsub = proplists:get_value(bisections, List, 0),
+    AutoScale = proplists:get_value(auto_scale_fit, List, false),
+    SetScale_S = proplists:get_value(set_scale, List, "100pt"),
+    UseViewboxCoords = proplists:get_value(use_viewbox_coords, List, false),
+    SetViewBoxScale = proplists:get_value(set_viewbox_scale, List, 0.010),
+    
+    AutoCenter = proplists:get_value(auto_center, List, true),
+    RemoveNonClosed = proplists:get_value(remove_nonclosed, List, true),
+    AllPathsInvertOverlap = proplists:get_value(all_paths_invert_overlap, List, false),
+    TransformsInLayerNames = proplists:get_value(transforms_in_layer_nms, List, false),
+    Props = [{extensions,[
+        {".svg",?__(4,".svg File")},
+        {".svgz",?__(8,"Compressed .svg File")}]}],
+    
+    wpa:import(Props, fun(F) ->
+        make_svg(F, #svg_importer_params{
+            nsubdiv=Nsub,
+            auto_scale=AutoScale,
+            set_scale=SetScale_S,
+            use_viewbox_c=UseViewboxCoords,
+            set_viewbox_s=SetViewBoxScale,
+            auto_center=AutoCenter,
+            remove_non_cls=RemoveNonClosed,
+            all_paths_invert=AllPathsInvertOverlap,
+            transforms_in_layers=TransformsInLayerNames
+        })
+    end, St);
 command(_, _) ->
     next.
 
-make_svg(Name, Nsubsteps) ->
-    case catch try_import_svg(Name, Nsubsteps) of
+make_svg(Name, #svg_importer_params{
+    nsubdiv=Nsubsteps,
+    auto_scale=AutoScale,
+    set_scale=SetScale_S,
+    use_viewbox_c=UseViewboxCoords,
+    set_viewbox_s=SetViewBoxScale,
+    auto_center=AutoCenter,
+    remove_non_cls=RemoveNonClosed,
+    all_paths_invert=AllPathsInvertOverlap,
+    transforms_in_layers=TransformsInLayerNames
+}=_)
+->
+    case parse_float_number_w_unit(SetScale_S, 0.0) of
+        {ScaleVal, ScaleUnit} when ScaleVal > 0.001 ->
+            wpa:pref_set(wpc_svg_path, svg_set_scale, SetScale_S),
+            SetScale = {ScaleVal, unit_atom(ScaleUnit)};
+        _Unk ->
+            SetScale = {100.0, pt}
+    end,
+    case UseViewboxCoords of
+        true  -> SetViewBoxScale_1 = SetViewBoxScale;
+        false -> SetViewBoxScale_1 = false
+    end,
+    case catch try_import_svg(Name, Nsubsteps, AutoScale, SetScale,
```

Comment:
Would prefer a `try catch E:R:ST ->` construct here, we try to avoid the old "catch" construct.